DataConverters3D/Object3D/Object3DExtensions.cs

```diff
@@ -207,6 +221,32 @@ public static IEnumerable<IObject3D> Descendants(this IObject3D root)
 				yield return item;
 				foreach (var n in item.Children)
 				{
+					// This is code that ensures the tree is bulid with parent pointers correctly
+					n.Parent = item;
```

Comment:
It doesn't seem like we should be assigning parent on tree iteration. 

I'd rather get Parent assignment right than have hacks in place that set parent if you iterate via Descendants but not otherwise. i.e. we're masking some problem with this what won't be evident until I write code that doesn't call these Descendants* functions